server/src/main/java/org/eclipse/openvsx/UserAPI.java

```diff
@@ -152,4 +156,105 @@ public ResultJson deleteAccessToken(@PathVariable("id") long id) {
         return ResultJson.success("Deleted access token for user " + user.getLoginName());
     }
 
+    @GetMapping(
+        path = "/user/namespaces",
+        produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    public List<NamespaceJson> getOwnNamespaces() {
+        var principal = users.getOAuth2Principal();
+        if (principal == null) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN);
+        }
+        var user = users.updateUser(principal);
+
+        var memberships = repositories.findMemberships(user);
+
+        return memberships.map(membership -> {
+            var namespace = membership.getNamespace();
+            var json = new NamespaceJson();
+            json.name = namespace.getName();
+            json.extensions = new LinkedHashMap<>();
+            var serverUrl = UrlUtil.getBaseUrl();
+            for (var ext : repositories.findExtensions(namespace)) {
+                String url = createApiUrl(serverUrl, "api", namespace.getName(), ext.getName());
+                json.extensions.put(ext.getName(), url);
+            }
+            var ownerships = repositories.findMemberships(namespace, NamespaceMembership.ROLE_OWNER);
+            json.access = ownerships.isEmpty() ? NamespaceJson.PUBLIC_ACCESS : NamespaceJson.RESTRICTED_ACCESS;
+
+            return json;
+        }).toList();
+    }
+
+    @GetMapping(
+        path = "/user/members/namespace/{name}",
+        produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    public List<NamespaceMembershipJson> getNamespaceMembers(@PathVariable String name) {
+        var principal = users.getOAuth2Principal();
+        if (principal == null) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN);
+        }
+        var namespace = repositories.findNamespace(name);
+        var memberships = repositories.findMemberships(namespace);
+
+        return memberships.map(membership -> membership.toJson()).toList();
+    }
+
+    @GetMapping(
+        path = "/user/{name}",
```

Comment:
Maybe just reuse the `/user/search/{name}` endpoint? This would also resolve the problem mentioned in my comment to UserDataRepository (at least from the server's perspective).